### DISCUSSION TITLE
Improve the queries directory to better fit w/ USQL

### DISCUSSION THREAD

**Discussion by mcrascal** [evidence-dev/evidence#2406] [General]
## Feature Description

The details of the `queries/` directory are a bit of a hold-over from pre USQL. The way they work now is kind of incongruent w/ the mental model of USQL, and takes you down a path of writing really non-performant pages. 

Queries in the queries directory help you accomplish two things: 

1. Cross source: Light data modelling across all your sources (e.g. joining the results of a postgres query to a snowflake query)
2. Building light pipelines (e.g. a small dag of queries) 
3. Quick refactor target for queries you're repeating in pages. Copy/paste, keep the same syntax. Even if you would eventually want to full refactor the query into your source. 

Today, there are two issues 

1. "Importing" queries in front matter, and having them rendered on the page encourages users to put very large unaggregated queries on the page, which is a performance drag 
2. You need to have a query on the page in order to use another one which references it. 

Here's how I think they should work: 

1. Frontmatter references should not be required 
2. Queries in `queries/` should only be able to reference sources and other queries in the directory. 
3. When we're compiling a chained query on a page, we should first check on the page for a query of that name, then check the queries directory, and swap in the compiled results from the queries directory.

**Comment by archiewood** [comment]
I'd add one goal I've seen people use file queries for:
- Separating out data aggregation from the markup: I think in some, (especially more complex) layouts, having the SQL query in the markdow can make it harder to understand what is going to be displayed.

**Comment by archiewood** [comment]
we have discussed calling this directory `views`
